lib/routes/fiat_currencies/fiat_currency_settings.dart

```diff
@@ -0,0 +1,274 @@
+import 'dart:math';
+
+import 'package:breez/bloc/account/account_actions.dart';
+import 'package:breez/bloc/account/account_bloc.dart';
+import 'package:breez/bloc/account/account_model.dart';
+import 'package:breez/bloc/account/fiat_conversion.dart';
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/user_profile/breez_user_model.dart';
+import 'package:breez/bloc/user_profile/fiat_currency_preferences.dart';
+import 'package:breez/bloc/user_profile/user_actions.dart';
+import 'package:breez/bloc/user_profile/user_profile_bloc.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/back_button.dart' as backBtn;
+import 'package:breez/widgets/flushbar.dart';
+import 'package:breez/widgets/loader.dart';
+import 'package:collection/collection.dart';
+import 'package:dragable_flutter_list/dragable_flutter_list.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+class FiatCurrencySettings extends StatefulWidget {
+  FiatCurrencySettings();
+
+  @override
+  FiatCurrencySettingsState createState() {
+    return FiatCurrencySettingsState();
+  }
+}
+
+class FiatCurrencySettingsState extends State<FiatCurrencySettings> {
+  AccountBloc _accountBloc;
+  AccountModel _accountModel;
+  UserProfileBloc _userProfileBloc;
+  List<String> _preferredFiatCurrencies;
+  List<FiatConversion> _fiatConversionList;
+
+  List<FiatConversion> _selectedFiatConversions;
+  List<FiatConversion> _unselectedFiatConversions;
+
+  bool _isInit = false;
+
+  @override
+  void didChangeDependencies() {
+    if (!_isInit) {
+      _accountBloc = AppBlocsProvider.of<AccountBloc>(context);
+      _userProfileBloc = AppBlocsProvider.of<UserProfileBloc>(context);
+      _getExchangeRates();
+      _getUserFiatCurrencyPreferences();
+      _initializeFiatCurrencyList();
+
+      _isInit = true;
+    }
+    super.didChangeDependencies();
+  }
+
+  void _getExchangeRates() {
+    FetchRates fetchRatesAction = FetchRates();
+    _accountBloc.userActionsSink.add(fetchRatesAction);
+    fetchRatesAction.future.catchError((err) {
+      if (this.mounted) {
+        setState(() {
+          Navigator.pop(context);
+          showFlushbar(context,
+              message: "Failed to retrieve BTC exchange rate.");
+        });
+      }
+    });
+  }
+
+  void _getUserFiatCurrencyPreferences() {
+    _userProfileBloc.userStream.firstWhere((user) => user != null).then((user) {
+      _preferredFiatCurrencies =
+          List.from(user.fiatCurrencyPreferences.preferredFiatCurrencies);
+    });
+  }
+
+  void _initializeFiatCurrencyList() {
+    _accountBloc.accountStream
```

Comment:
Seems better to use Stream builder instead of fetching and sorting on init.

Reply:
[Done](https://github.com/breez/breezmobile/pull/394/commits/d6c942f95931ca79fd135292915b3a08c07d56aa)

Sort logic is moved to bloc.